feat: Add `domain_name` output

RoseSecurity opened the pull request:
## Description

This adds a new `domain_name` output to expose the domain name attribute from the OpenSearch domain resource, allowing users to reference the domain name in their Terraform configurations for constructing endpoints, configuring DNS records, or referencing the domain in other AWS resources.

## Motivation and Context

- Closes #46 

## Breaking Changes

N/A

## How Has This Been Tested?

- [X] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [X] I have tested and validated these changes using one or more of the provided `examples/*` projects
- [X] I have executed `pre-commit run -a` on my pull request

RoseSecurity committed:
feat(opensearch): update TLS policy and add domain_name output

- Changed the default TLS security policy for the OpenSearch domain to
  "Policy-Min-TLS-1-2-PFS-2023-10" for improved security.
- Added a new output "domain_name" to expose the OpenSearch domain name.

RoseSecurity committed:
chore: update docs

RoseSecurity committed:
chore: update examples

bryantbiggs commented on main.tf:
```diff
@@ -179,7 +179,7 @@ resource "aws_opensearch_domain" "this" {
       custom_endpoint_enabled         = try(domain_endpoint_options.value.custom_endpoint_enabled, null)
       enforce_https                   = try(domain_endpoint_options.value.enforce_https, true)
-      tls_security_policy             = try(domain_endpoint_options.value.tls_security_policy, "Policy-Min-TLS-1-2-2019-07")
+      tls_security_policy             = try(domain_endpoint_options.value.tls_security_policy, "Policy-Min-TLS-1-2-PFS-2023-10")
```
this would be a breaking change so lets skip this for now - feel free to open an issue to have this changed in the next major version

RoseSecurity replied:
Just made the changes! Thanks!

RoseSecurity committed:
fix: update default TLS policy to Policy-Min-TLS-1-2-PFS-2023-10

Updated the default value for `tls_security_policy` in domain endpoint
options from "Policy-Min-TLS-1-2-2019-07" to
"Policy-Min-TLS-1-2-PFS-2023-10" in variables, wrapper, and documentation.
Also added `domain_name` output to the complete example README for
consistency.

RoseSecurity committed:
fix(domain): update default TLS policy to 2019-07

Change the default value for `tls_security_policy` in domain endpoint
options from "Policy-Min-TLS-1-2-PFS-2023-10" to
"Policy-Min-TLS-1-2-2019-07" across documentation, variables, and
module usage. This ensures compatibility with a broader range of
clients and aligns with AWS recommended defaults.

bryantbiggs reviewed (approved):
thank you!

antonbabenko commented:
This PR is included in version 2.3.0 :tada:
